src/Http/Client/Adapter/Stream.php

```diff
@@ -147,15 +147,19 @@ protected function _buildHeaders(Request $request, $options)
     {
         $headers = [];
         foreach ($request->getHeaders() as $name => $values) {
-            $headers[] = sprintf('%s: %s', $name, implode(", ", $values));
+            if ($name !== 'Cookie') {
+                $headers[] = sprintf('%s: %s', $name, implode(", ", $values));
+            }
         }
 
+        $cookieHeader = $request->getHeaderLine('Cookie');
         $cookies = [];
         foreach ($request->cookies() as $name => $value) {
             $cookies[] = "$name=$value";
         }
-        if ($cookies) {
-            $headers[] = 'Cookie: ' . implode('; ', $cookies);
+        $cookieData = implode('; ', $cookies);
+        if ($cookieData || $cookieHeader) {
+            $headers[] = 'Cookie: ' . implode('; ', [$cookieHeader, $cookieData]);
```

Comment:
Just realised I am missing a test​ where a cookie is both in the jar and defined in options. I will add that.